e2e/launcher/network.go

```diff
@@ -0,0 +1,43 @@
+package launcher
+
+import (
+	"fmt"
+	"log"
+	"os"
+)
+
+type Network struct {
+	Nodes                []*Node
+	NetworkConfiguration *NetworkConfiguration
+}
+
+func LaunchNetwork(networkConfigDirectory string, executablePath string) (network Network, err error) {
+	fmt.Println("./launcher/network_configs/" + networkConfigDirectory)
+	networkConfiguration, err := loadNetworkConfiguration("../launcher/network_configs/" + networkConfigDirectory)
+	if err != nil {
+		return
+	}
+	networkRootDirectory, err := os.MkdirTemp("", networkConfiguration.NetworkId+"-")
+	if err != nil {
+		return
+	}
+	log.Printf("root network directory: %v", networkRootDirectory)
+	var nodes []*Node
+	for _, nodeConfiguration := range networkConfiguration.NodeConfigurations {
+		var node *Node
+		node, err = newNode(nodeConfiguration, networkRootDirectory, networkConfiguration.GenesisPath, executablePath)
```

Comment:
This is a no-no: in the worst case scenario changing code to be like this would shadow the return value. In this particular case, it would merely cause a compilation error.
Here, two examples:
- [here](https://go.dev/play/p/399zRdoXppd) for the shadowing case (run and look at the logs).
- [here](https://go.dev/play/p/86efKsyifB7) for the compilation error case